### PR TITLE
updated .gitignore

### PR DESCRIPTION
**Pull request by ohbus**
For Eclipse, IntelliJ and NetBeans

Fixes #

## Proposed Changes

- Added .gitignore for Eclipse and NetBeans Development
- Updated .gitignore for Idea development

 🧽 Update or clean up current behavior

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,35 @@
...
+!**/src/main/**
+!**/src/test/**
+
+### STS ###
+**/.apt_generated
+**/.classpath
+**/.factorypath
+**/.project
+**/.settings
+**/.springBeans
```
Comment (slinkydeveloper) [inline .gitignore:17]: springBeans?

File: .gitignore [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
-
 **/target
 
-**/*.iml
-
 **/dependency-reduced-pom.xml
 
 **/?
+
+!.mvn/wrapper/maven-wrapper.jar
```
[the remark this answers is not in the document]
Comment (ohbus) [reply .gitignore:7]: I'll remove these

File: .gitignore [hunk 3]
```diff
@@ -1,9 +1,35 @@
...
+!**/src/main/**
+!**/src/test/**
```
[the remark this answers is not in the document]
Comment (ohbus) [reply .gitignore:9]: No problem, I'll remove this

File: .gitignore (hunk 1 above)
Comment (ohbus) [reply .gitignore:17]: Actually I was using Eclipse and these are for my Spring projects and just added them.

I'll get rid of factorypath and springbeans for the meantime

**Comment by ohbus** [comment]
@slinkydeveloper and @pierDipi I have made the changes as requested.

**Comment by pierDipi** [comment]
@ohbus Thanks and welcome to Knative Eventing Kafka Broker :tada:

**Pull request merged**